### PR TITLE
Import existing organisation information

### PR DESCRIPTION
**Pull request by ramyaragupathy**
Includes a migration file to:
* [x] import all existing organisation tag into new organisation table
* [x] Match the ID from organisation table with projects

### CODE REVIEW

**Comment by willemarcel** [comment]
@ramyaragupathy there are some `black` alerts to solve.

**Comment by ramyaragupathy** [comment]
One question here @xamanu - how do we capture manager information for new organsiations populated from existing project info? There are multiple projects from different authors mapped to  one organisation. Looks like we have a constraint - one manager per org in our table. Should we just use the first author id encountered for an org as the manager?

**Comment by xamanu** [comment]
That is a good question. I think we should reach out to the organizations we know to ask them who will be their manager. Let's discuss how many organizations we have and who are the people that created projects on their behalf. I hope this is still a manageable list.

**Comment by xamanu** [comment]
Quick status report. We are actively going through all the organizations and finding duplicates, grouping some of them, reaching out to contact persons, etc. This will take another week approx. Then we can continue here to bring this data in.

File: backend/models/postgis/project.py [hunk 1]
```diff
@@ -169,7 +169,7 @@ class Project(db.Model):
         db.ForeignKey("organisations.id", name="fk_organisations"),
         index=True,
     )
-    organisation_tag = db.Column(db.String, index=True)
+    # organisation_tag = db.Column(db.String, index=True)
```
Comment (xamanu) [inline backend/models/postgis/project.py]: @ramyaragupathy do we want/need this line?

**Comment by ramyaragupathy** [comment]
During upgrade:

* Identify all distinct `organisation_tag` from the `projects` table

_for loop start_

*  Check if there is mapping available for the tag within the migration file. If there is some mapping pick the mapped name as the new org name, else retain the original tag as the org name
* Check if there is already an organization entry existing for the org name. If it is not available create a new entry and pick the ID for the new organisation inserted into the table
* On projects table update organisation_id using organisation_tag info
* Now for the newly created org select the first matching author ID from associated projects. Use this info to create an org manager for the org under `organisation_managers`

_for loop end_

* After all the existing organisation_tag has been mapped to organisation and projects, display finish message to users

During downgrade:
* Add `organisation_tag` column to projects
* Select all existing organisation ID and name. Using this populate `organisation_tag` on projects with respective organisation name
*  Delete all entries in:
   * project_teams
   * team_members
   * teams
   * organisation_managers
   * set organisation_id field to null in projects table
   * nullify contents in organisation table

**Comment by ramyaragupathy** [comment]
Checklist after upgrade:
* Make sure organisations table is populated with new organisation names. 
* There should be a `Other` organisation - in total this migration introduces about 88 new orgs
*  `organisation_tag` column should have gone from `projects` and a new `organisation_id` with IDs from `organisation` should be in place
* Check for `organisation_managers` table and check the mapping between organisation & managers

On downgrade:
* empty `organisations` 
* `organisation_tag` field re-introduced on `projects` with organisation name populated from `organisations`
* empty `teams`
* empty `organisation_managers`
* empty `team_members`

**Comment by willemarcel** [comment]
That migration conflicted with the one in #2518, because both referenced `c40e1fdf6b70` as the parent. I updated on my machine the migration of this PR to run after `14340f1e0d6b`, but it failed with:

```
INFO  [alembic.runtime.migration] Running upgrade c40e1fdf6b70 -> 14340f1e0d6b, empty message
INFO  [alembic.runtime.migration] Running upgrade 14340f1e0d6b -> 7bbc01082457, empty message
Populating organisation information for Projects....
Total distinct organisations in the DB: 4
Traceback (most recent call last):ts
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1247, in _execute_context
    self.dialect.do_execute(
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 588, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "organisation_user_key"
DETAIL:  Key (organisation_id, user_id)=(18, 360183) already exists.


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 144, in <module>
    manager.run()
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/flask_script/__init__.py", line 417, in run
    result = self.handle(argv[0], argv[1:])
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/flask_script/__init__.py", line 386, in handle
    res = handle(*args, **config)
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/flask_script/commands.py", line 216, in __call__
    return self.run(*args, **kwargs)
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/flask_migrate/__init__.py", line 96, in wrapped
    f(*args, **kwargs)
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/flask_migrate/__init__.py", line 271, in upgrade
    command.upgrade(config, revision, sql=sql, tag=tag)
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/alembic/command.py", line 298, in upgrade
    script.run_env()
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/alembic/script/base.py", line 489, in run_env
    util.load_python_file(self.dir, "env.py")
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/alembic/util/pyfiles.py", line 98, in load_python_file
    module = load_module_py(module_id, path)
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/alembic/util/compat.py", line 184, in load_module_py
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "migrations/env.py", line 106, in <module>
    run_migrations_online()
  File "migrations/env.py", line 98, in run_migrations_online
    context.run_migrations()
  File "<string>", line 8, in run_migrations
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/alembic/runtime/environment.py", line 846, in run_migrations
    self.get_context().run_migrations(**kw)
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/alembic/runtime/migration.py", line 520, in run_migrations
    step.migration_fn(**kw)
  File "/home/wille/projects/HOT/tasking-manager/migrations/versions/7bbc01082457_.py", line 247, in upgrade
    conn.execute(
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 976, in execute
    return self._execute_text(object_, multiparams, params)
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1145, in _execute_text
    ret = self._execute_context(
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1287, in _execute_context
    self._handle_dbapi_exception(
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1481, in _handle_dbapi_exception
    util.raise_(
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/sqlalchemy/util/compat.py", line 178, in raise_
    raise exception
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 1247, in _execute_context
    self.dialect.do_execute(
  File "/home/wille/.virtualenvs/taskman/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 588, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.IntegrityError: (psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "organisation_user_key"
DETAIL:  Key (organisation_id, user_id)=(18, 360183) already exists.

[SQL: insert into organisation_managers                     (organisation_id,user_id)                     values(18,360183)]
(Background on this error at: http://sqlalche.me/e/gkpj)
```

**Pull request merged**